### PR TITLE
Add the new feature to enable the new product editor blocks experience

### LINKED ISSUE #37128: Create new dev feature flag for the product block work
**Issue Description:**

We want the product block work set behind a feature flag for now, but not one that can be enabled by a merchant yet as the editor is not in a working state yet. 
Once in a working state this will be replaced by an official feature flag (shown in the features settings page) as part of this issue: #36989

This new feature flag will be toggleable through the WooCommerce Beta tester plugin.

Current logic:
**No features enabled** -> Displays legacy product editor
**New Product Editor** (enabled through **Settings > Advanced > Features**) -> Displays new product editor (non block version)
**Block editor feature enabled** -> Displays the new product block editor ( trumps both the above ).

**Acceptance criteria**

- [x] A new key should be available for the product block work in both `core.json` & `development.json` here: https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/client/admin/config (enabled in `development`)
- [x] If the new block feature is enabled it should load the product block editor
- [x] If the block feature is disabled it should show the old editor, or the first iteration of the product editor if the feature is enabled within WooCommerce settings
- [x] This feature should be toggleable through the beta tester features tab.

### PR DESCRIPTION
**Pull request by mdperez86**
### All Submissions:

-   [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/37128

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Make sure you have `WooCommerce Admin Test Helper` plugin installed.
2. Visit `/wp-admin/tools.php?page=woocommerce-admin-test-helper`.
3. Under `Features` tab if you enable the feature `block-editor-feature-enabled` and then visit the page `Products -> Add New` you should see the new product editor blocks experience.
4. If you enable both features `new-product-management-experience` and `block-editor-feature-enabled` you should see the same thing as in point 3.
5. If you disable `block-editor-feature-enabled` while keep `new-product-management-experience` you should see the Product Form (non legacy) experience.
6. If you disable both you should see the legacy experience.

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [ ] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/src/Admin/Features/BlockEditorFeatureEnabled.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	/**
+	 * Option name used to toggle this feature.
+	 */
+	const TOGGLE_OPTION_NAME = 'woocommerce_' . self::FEATURE_ID . '_enabled';
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		if ( ! Features::is_enabled( 'new-product-management-experience' ) && Features::is_enabled( self::FEATURE_ID ) ) {
```
Comment (joelclimbsthings) [inline plugins/woocommerce/src/Admin/Features/BlockEditorFeatureEnabled.php:28]: So, if both feature flags are enabled, the block editor is still displayed but this condition is not met, so the styles/etc are not enqueued. Should this just be:
```php
if (Features::is_enabled( self::FEATURE_ID ) )
```

?

**Review by joelclimbsthings — CHANGES_REQUESTED** [review]
Thanks @mdperez86 , this will be a big help. I had one question, but aside from that  in my testing it seems to be working well!

File: plugins/woocommerce/src/Admin/Features/BlockEditorFeatureEnabled.php (hunk 1 above)
Comment (mdperez86) [reply plugins/woocommerce/src/Admin/Features/BlockEditorFeatureEnabled.php:28]: Yes they are. The same styles are being enqueued also if `new-product-management-experience` is enabled. So we don't want the same styles to be enqueue twice. See `plugins/woocommerce/src/Admin/Features/NewProductManagementExperience.php`.
Good catch!!. But this is intentional.

Comment (joelclimbsthings) [reply plugins/woocommerce/src/Admin/Features/BlockEditorFeatureEnabled.php:28]: Ah, understood!

**Review by joelclimbsthings — APPROVED** [review]
Great job @mdperez86 , looks good! 🚢

**Pull request merged**